### PR TITLE
Temperature: Update flash interface

### PR DESCRIPTION
**Pull request by matias-cphatomics**
- Tested on Temperature PCB V6.0

**Review Changes**
 - Update CA_Embedded_Libraries to succesfully run unit tests
 - Update of the unitTests.py file to fail workflow if cmake fails to build the tests
 - Error message if flash write fails

### CODE REVIEW

File: STM32/Temperature/Core/Src/Temperature.c [hunk 1]
```diff
@@ -279,7 +278,9 @@ static void calibrateTypeInput(int noOfCalibrations, const CACalibration* calibr
 static void calibrateReadWrite(bool write)
 {
     if (write)
-        writeToFlashSafe(hcrc, 0, sizeof(portCalVal), (uint8_t *) portCalVal);
+    {
+        writeToFlashCRC(hcrc, (uint32_t) FLASH_ADDR_CAL, (uint8_t *) portCalVal, sizeof(portCalVal));
```
Comment (TimotheDodinCA) [inline STM32/Temperature/Core/Src/Temperature.c:282]: On the oxygen board, you did this to print a message if an error occurs:

`if (writeToFlashCRC(hcrc_, (uint32_t) FLASH_ADDR_CAL, (uint8_t *) &cal, sizeof(cal)) != 0)
        {
            USBnprintf("Calibration was not stored in FLASH");
        }`

**Comment by LukeWalker-CA** [comment]
The code looks fine, but the unit tests fail. The YML hasn't caught the error, which is concerning

**Comment by matias-cphatomics** [comment]
> The code looks fine, but the unit tests fail. The YML hasn't caught the error, which is concerning

It looks like it fails because the fake_ADS1120 has not been merged yet  from this [branch](https://github.com/copenhagenatomics/CA_Embedded_Libraries/tree/dev/temp6.1-testing). I assume the .yml also failed when the latest temperature PR was merged into main. 

@LukeWalker-CA will you have a look as to whether the branch is ready for review / merging?

**Comment by LukeWalker-CA** [comment]
Opened a PR [here](https://github.com/copenhagenatomics/CA_Embedded_Libraries/pull/57)

**Review by TimotheDodinCA — APPROVED** [review]

**Pull request merged**